dotnet/src/Microsoft.Agents.AI.AGUI/Shared/AGUIChatMessageExtensions.cs

```diff
@@ -44,5 +207,44 @@ public static ChatRole MapChatRole(string role) =>
         string.Equals(role, AGUIRoles.User, StringComparison.OrdinalIgnoreCase) ? ChatRole.User :
         string.Equals(role, AGUIRoles.Assistant, StringComparison.OrdinalIgnoreCase) ? ChatRole.Assistant :
         string.Equals(role, AGUIRoles.Developer, StringComparison.OrdinalIgnoreCase) ? s_developerChatRole :
+        string.Equals(role, AGUIRoles.Tool, StringComparison.OrdinalIgnoreCase) ? ChatRole.Tool :
         throw new InvalidOperationException($"Unknown chat role: {role}");
+
+    private static JsonElement ConvertToJsonElement(object value, JsonSerializerOptions jsonSerializerOptions)
```

Comment:
cc: @eiriktsarpalis... this looks like a lot of ceremony that shouldn't be necessary?